.circleci/config.yml

```diff
@@ -56,13 +56,16 @@ jobs:
           name: Push application Docker image
           command: |
             docker login -u ${QUAY_USERNAME} -p ${QUAY_PASSWORD} quay.io
+            docker tag app "quay.io/mojanalytics/control-panel:${CIRCLE_SHA1}"
             docker tag app "quay.io/mojanalytics/control-panel:${CIRCLE_BRANCH}"
+            docker push "quay.io/mojanalytics/control-panel:${CIRCLE_SHA1}"
             docker push "quay.io/mojanalytics/control-panel:${CIRCLE_BRANCH}"
 
       - deploy:
```

Comment:
Question: Doesn't this mean we always deploy after tests pass? So in theory the environment or branch shouldn't matter from the CD perspective? I may miss something

Reply:
Yes, continuous deployment basically means deploying when tests pass. What this essentially does is create an environment for every branch.